mvrx/src/main/kotlin/com/airbnb/mvrx/MvCorStateStore.kt

```diff
@@ -0,0 +1,115 @@
+package com.airbnb.mvrx
+
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import kotlinx.coroutines.experimental.CoroutineDispatcher
+import kotlinx.coroutines.experimental.DefaultDispatcher
+import kotlinx.coroutines.experimental.channels.Channel
+import kotlinx.coroutines.experimental.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.experimental.channels.actor
+import kotlinx.coroutines.experimental.rx2.asObservable
+import java.util.*
+
+/**
+ * This is a container class around the actual state itself. It has a few optimizations to ensure
+ * safe usage of state.
+ *
+ * All state reducers are run in an actor to ensure that they don't have race
+ * conditions with each other.
+ *
+ */
+
+internal open class MvCorStateStore<S : Any>(initialState: S, coroutineDispatcher: CoroutineDispatcher = DefaultDispatcher) : Disposable, IMvRxStateStore<S> {
```

Comment:
If my understanding is correct, it doesn't matter for actors, since actor is a coroutine and is executed sequentially. Here is [relevant documentation](https://github.com/Kotlin/kotlinx.coroutines/blob/master/coroutines-guide.md#actors). It might make sense to create a separate cached pool though, because Default can be starved by other tasks